inst/include/sampler.h

```diff
@@ -161,6 +162,15 @@ namespace smc {
         historyflags GetHistoryFlags(long n) const {return History[n].GetFlags();}
         /// Returns the Effective Sample Size of the specified particle generation.
         double GetESS(long n) const {return History[n].GetESS();}
+        /// Returns a vector of population objects re-ordered to match ancestral lines of the particle system
+        std::vector<population<Space>> GetApop(void) const;
+        // void GetALineInd(long n) const;
+        arma::Col<unsigned int> GetALineInd(long n) const;
+        std::vector<Space> GetALineSpace(long n) const;
+        /// HELPER FUNCTION THAT MAY BE DELETED; RETURNS uRSIndices

```

Comment:
Are these functions intended to be used externally? If not should they be private or protected?

It just seems a bit unfriendly to out to the interface something with a "MAY BE DELETED" comment attached. I don't immediately see any harm in leaving these functions but if we do I'd be inclined to do that with the view to keeping them permanently (or as permanently as one does anything in code).

Reply:
1. I think that depends on how we want to implement conditional SMC and grant the user access to these class members. 
* It would be more flexible to allow the user access to all trajectories, hence I'd suggest `GetALineSpace()` should be public.
* I believe that `GetALineSpace()` is only ever useful for internal purposes, specifically, also for derived cond. SMC. So `GetALineInd()` should be protected. 

If no objections I'll do as above.

2. Ah I see, my mistake, just wanted to highlight this with capslock for a discussion.
* I think, we might want to decide whether to make ancestral lines accessible only via the historyelement class (via `historyelement.SetAindices()`) or additionally (but this might be an unnecessary deflection since it only gives access to the current iteration resampled indices) via 
https://github.com/rcppsmc/rcppsmc/blob/6bd4ceaa976d9285cb1d867515bb003e0f3d7361/inst/include/sampler.h#L170-L171

* The same goes for 
https://github.com/rcppsmc/rcppsmc/blob/6bd4ceaa976d9285cb1d867515bb003e0f3d7361/inst/include/sampler.h#L172-L173

Maybe there are other merits to these functions, that I don't see. In general, I'd be a bit more inclined to delete unnecessary facilities (only keeping what's relevant, for clarity purposes). 

What do the others think @eddelbuettel @LeahPrice ?

Reply:
I'll interrupt given I raised this in the first place.

Presumably in 1. the `GetALineSpace` in the second bullet point is a typo for `GetALineInd`. No objections from me, although I didn't want to insist on anything just that if we have functions which we might get rid of them exposing their interface might not be helpful. I still have no problem with keeping them; it's difficult to predict every possible use case and making everything available to users that might want it doesn't seem to hurt.

No strong views either way on 2. Largely a matter of taste, I think.

Reply:
I also don't think we should make functions public unless we plan on keeping them, and I'm not too concerned about keeping functions that we can't see a direct use for right now if they might be useful later. I don't have any strong opinions on point 2.